include/git2/sys/textconv.h

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#ifndef INCLUDE_sys_git_textconv_h__
+#define INCLUDE_sys_git_textconv_h__
+
+#include "git2/textconv.h"
+
+/**
+ * @file git2/sys/textconv.h
+ * @brief Git textconv backend and plugin routines
+ * @defgroup git_backend Git custom backend APIs
+ * @ingroup Git
+ * @{
+ */
+GIT_BEGIN_DECL
+
+/**
+ * Look up a textconv by name
+ *
+ * @param name The name of the textconv
+ * @return Pointer to the textconv object or NULL if not found
+ */
+GIT_EXTERN(git_textconv *) git_textconv_lookup(const char *name);
+
+#define GIT_TEXTCONV_HTTP  "http"
+
+/**
+ * Initialize callback on textconv
+ *
+ * Specified as `textconv.initialize`, this is an optional callback invoked
+ * before a textconv is first used.  It will be called once at most.
+ *
+ * If non-NULL, the textconv's `initialize` callback will be invoked right
+ * before the first use of the textconv, so you can defer expensive
+ * initialization operations (in case libgit2 is being used in a way that
+ * doesn't need the textconv).
+ */
+typedef int (*git_textconv_init_fn)(git_textconv *self);
+
+/**
+ * Shutdown callback on textconv
+ *
+ * Specified as `textconv.shutdown`, this is an optional callback invoked
+ * when the textconv is unregistered or when libgit2 is shutting down.  It
+ * will be called once at most and should release resources as needed.
+ * This may be called even if the `initialize` callback was not made.
+ *
+ * Typically this function will free the `git_textconv` object itself.
```

Comment:
"Typically" ? I take it means the callback is expected to free that object, right ? It might be clearer to make it a "must".